core/src/autogluon/core/utils/utils.py

```diff
@@ -435,20 +486,42 @@ def generate_train_test_split(
     """
     if len(X) == 1:
         raise ValueError(f"Cannot split data into train/val as it contains only one sample.")
-    if isinstance(test_size, float):
-        if (test_size <= 0.0) or (test_size >= 1.0):
-            raise ValueError("fraction of data to hold-out must be specified between 0 and 1")
-    elif isinstance(test_size, int):
-        assert test_size > 0
-    valid_problem_types = [BINARY, MULTICLASS, REGRESSION, SOFTCLASS, QUANTILE]
-    assert problem_type in valid_problem_types, f'Unknown problem type "{problem_type}" | Valid problem types: {valid_problem_types}'
+    if test_size is None and train_size is None:
+        test_size = 0.1
```

Comment:
qq: why do we go with a 90/10 and not 80/20

Reply:
Because I made that decision 5 or 6 years ago and now it is legacy. I felt that 90/10 is better than 80/20. No quantitative reason.